packages/@divvi/mobile/src/divviProtocol/register.ts

```diff
@@ -0,0 +1,32 @@
+import { getDataSuffix } from '@divvi/referral-sdk'
+import { getAppConfig } from 'src/appConfig'
+import { store } from 'src/redux/store'
+import Logger from 'src/utils/Logger'
+import { hasReferralSucceededSelector } from './slice'
+
+/**
+ * Checks if a user is referred to each provider and returns a data suffix for providers
+ * where the user is not yet referred.
+ *
+ * @returns A data suffix string for providers where the user is not yet referred, or empty string if no referrals needed
+ */
+export function getDivviData() {
+  const config = getAppConfig()
+  const consumer = config.divviProtocol?.divviId
+  const providers = config.divviProtocol?.campaignIds
+
+  if (!consumer || !providers || !providers.length) {
+    return null
+  }
+
+  // Check if this combination has already been successfully referred
+  const state = store.getState()
+  if (hasReferralSucceededSelector(state, consumer, providers)) {
```

Comment:
shouldn't this also check for pending? so there aren't multiple transactions for the same consumer + providers?

Reply:
Hmm, I don't think I am fully convinced this retry is worth it given we have a pretty robust retry system already. 
This also potentially leads to two events being tried at the same time if a new tx is submitted while one is in the middle of being processed. And also adds a little more gas cost to the user. But I'll leave it up to you. Can we add a comment on why we override pending events here?